crates/oxc_linter/src/rules/eslint/eqeqeq.rs

```diff
@@ -88,7 +88,7 @@ declare_oxc_lint!(
     Eqeqeq,
     eslint,
     pedantic,
-    conditional_fix
+    conditional_fix_dangerous,
```

Comment:
I am not sure how the fixer internally works. What happens when we use only safe fixes? Does the Fixer prefilter them and ignore the safe fixes?
Maybe we want to create some test for it?

Reply:
I'm not entirely sure what you mean: as far as this line goes, this is declaring that the rule _can_ emit dangerous fixes, but it is limited to _only_ emitting dangerous fixes. The safeness is per-fix, so even with just passing `oxlint --fix`, this will still fix safe `eqeqeq` violations, but you will need to pass `oxlint --fix --fix-dangerously` to fix all.

Reply:
> The safeness is per-fix, so even with just passing oxlint --fix, this will still fix safe eqeqeq violations, but you will need to pass oxlint --fix --fix-dangerously to fix all.

That is the expected behavior, I am not sure if that is working as intended :)